pkg/apiserver/experiment/experiment.go

```diff
@@ -67,11 +71,50 @@ func (s *Service) listExperiments(c *gin.Context) {}
 // TODO: need to be implemented
 func (s *Service) getExperimentDetail(c *gin.Context) {}
 
-// TODO: need to be implemented
-func (s *Service) deleteExperiment(c *gin.Context) {}
+func (s *Service) deleteExperiment(c *gin.Context) {
+	kind := c.Query("kind")
+	if kind == "" {
+		util.JSONEmptyData(c, statuscode.ParamNotSet, "kind param should be set")
+		return
+	}
+
+	ctx := context.TODO()
+	chaosKey := types.NamespacedName{Namespace: c.Param("ns"), Name: c.Param("name")}
+
+	kindsMap := map[string]runtime.Object{
+		v1alpha1.KindPodChaos:     &v1alpha1.PodChaos{},
+		v1alpha1.KindNetworkChaos: &v1alpha1.NetworkChaos{},
+		v1alpha1.KindStressChaos:  &v1alpha1.StressChaos{},
+		v1alpha1.KindIOChaos:      &v1alpha1.IoChaos{},
+		v1alpha1.KindKernelChaos:  &v1alpha1.KernelChaos{},
+		v1alpha1.KindTimeChaos:    &v1alpha1.TimeChaos{},
+	}
```

Comment:
Can we make this as a const variable?

Reply:
Const map is not possible. I will update it to use a var.